python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/ethernet-interfaces.j2

```diff
@@ -160,6 +160,47 @@ interface {{ ethernet_interface.name }}
 {%     if ethernet_interface.pvlan_mapping is arista.avd.defined %}
    switchport pvlan mapping {{ ethernet_interface.pvlan_mapping }}
 {%     endif %}
+{%     if ethernet_interface.vlan_id is arista.avd.defined and ethernet_interface.type | arista.avd.default != 'l2dot1q' %}
+   vlan id {{ ethernet_interface.vlan_id }}
+{%     endif %}
+{%     if ethernet_interface.encapsulation_dot1q.vlan is arista.avd.defined %}
+{%         set encapsulation_dot1q_cli = 'encapsulation dot1q vlan ' ~ ethernet_interface.encapsulation_dot1q.vlan %}
+{%         if ethernet_interface.encapsulation_dot1q.inner_vlan is arista.avd.defined %}
+{%             set encapsulation_dot1q_cli = encapsulation_dot1q_cli ~ ' inner ' ~ ethernet_interface.encapsulation_dot1q.inner_vlan %}
+{%         endif %}
+   {{ encapsulation_dot1q_cli }}
+{%     elif ethernet_interface.encapsulation_vlan.client.protocol is arista.avd.defined %}
+{%         set protocol = ethernet_interface.encapsulation_vlan.client.protocol %}
+{%         set network_flag = False %}
+{%         if protocol in ['dot1q', 'dot1ad'] %}
+{%             if ethernet_interface.encapsulation_vlan.client.vlan is arista.avd.defined %}
+{%                 set encapsulation_cli = 'client ' ~ protocol ~ ' ' ~ ethernet_interface.encapsulation_vlan.client.vlan %}
```

Comment:
I think you have not covered for this case.
`interface Ethernet26.400
   encapsulation vlan
      client dot1q outer 100 inner dot1ad 101`

Reply:
when protocol for outer and inner are not same.

Reply:
Added `inner_protocol` key